cpp/include/cudf/detail/gather.cuh

```diff
@@ -522,14 +521,18 @@ void gather_bitmask(table_view const& source,
   }();
   impl(*device_source,
        gather_map,
-       masks,
+       d_target_masks.data(),
        target.size(),
        target_rows,
-       d_valid_counts.data().get(),
+       d_valid_counts.data(),
        stream);
 
   // Copy the valid counts into each column
-  auto const valid_counts = thrust::host_vector<size_type>(d_valid_counts);
+  std::vector<size_type> valid_counts(d_valid_counts.size());
+  CUDA_TRY(cudaMemcpy(valid_counts.data(),
+                      d_valid_counts.data(),
+                      d_valid_counts.size() * sizeof(size_type),
+                      cudaMemcpyDefault));
```

Comment:
I thought we typically used `cudaMemcpyAsync` + `stream.synchronize()`.

Reply:
Please request changes so this doesn't get merged while I'm on vacation.

Reply:
This can be replaced with `make_std_vector_sync` now
https://github.com/rapidsai/cudf/blob/d1c324510a8a0c00de21503d4853a1f88c59b3b2/cpp/include/cudf/detail/utilities/vector_factories.hpp#L320

Something like:
```
auto valid_counts = make_std_vector_sync(d_valid_counts, stream);
```